### ISSUE TITLE
Pagination & Sorting ability while listing features

### ISSUE THREAD

**Issue by mik047** [featurehub-io/featurehub#676]
**Is your feature request related to a problem? Please describe.**
Currently, if the features grow beyond a screen size, it is difficult to navigate among the features.

**Describe the solution you'd like**
Would be nice to have a pagination ability to navigate the features on UI. 

**Describe alternatives you've considered**
Maybe following will do the trick.
- putting the pagination text/buttons on top right corner of the feature listing
-- the text/buttons are to be auto enabled/disabled based on the parameters such as current page, page size, total count 
- the feature listing (or the page size) defaults to x (say x=5) 
- The listing is always sorted as per the natural understanding for most users (the dictionary style alphabetically descending text)

<img width="1788" alt="Screenshot 2022-01-26 at 12 03 04 PM" src="https://user-images.githubusercontent.com/5581890/151116017-f127d579-cd8e-4cad-83de-62979a61ca6f.png">

**Comment by cspalding** [comment]
Wanted to upvote this issue. My organization is experiencing the same problem. Another option aside from pagination would be a text input at the top of the screen to filter flags whose key/name match the text input.

**Comment by mik047** [comment]
Is that similar to https://github.com/featurehub-io/featurehub/issues/675?

**Comment by cspalding** [comment]
Ah yeah, you're right it is #675! I just hadn't seen that issue.

**Comment by ilyatovbin-pp** [comment]
needing this as well. both paginated pages and a way to search/filter flags would be appreciated.

**Comment by IrinaSouth** [comment]
Hey all, just an update that I made some progress on this and finally looking good with some redesign tweaks.. hoping to release by the end of 2022! :)  @ilyatovbin-pp @cspalding @mik047 

<img width="1155" alt="Screenshot 2022-12-13 at 20 17 38" src="https://user-images.githubusercontent.com/9538697/207251258-fa41b999-ce4c-4e8c-bc5d-ad3d6b7a36ef.png">

**Comment by IrinaSouth** [comment]
any feedback welcome of course :)

**Comment by mik047** [comment]
This looks neat. So if I understand, there is...

- (top) left most panel is to filter out environments
- (top) middle panel is to filter out type of features
- (top) right most panel is _fuzzy_ search to find features
  - I am assuming the search considers feature name. 
  - Does it _also_ consider feature key OR feature description? Well I understand, it will not give visual feedback like it will give for feature names but just asking.
- (bottom) panel provides the pagination + number of features per page
- (I am assuming), the display order of toggles remains in alphabetical order.
  -  I see nothing to change the sorting order but honestly I do not think it is required any more given the above features.

Thanks @IrinaSouth - looking forward to those changes in upcoming release. :)

**Comment by IrinaSouth** [comment]
thanks @mik047 !

[Q] Can the search also include feature [key and/or description]?  - currently search will return if search term contained in the name or key. We could expand this to add description too. 
I see no icon to alter sorting order.  - Sorting I believe is by key in Ascending order. Might be better to order by name? I'll see if I can add sorting button too. 

Cheers for the feedback!

**Comment by IrinaSouth** [comment]
Hey @mik047 @ilyatovbin-pp @cspalding - the new dashboard has been released in 1.5.11-RC. Ideally would be great if you give it a go and let me know if there are any issues with it. And then we can cut 1.5.11 release asap.  Thanks a lot!

**Comment by cspalding** [comment]
@IrinaSouth I just deployed the RC to my test environment. Wow! What an improvement. Page load times are way down and the filtering and sorting work seamlessly. Thanks a ton for all of your work on the updates! I've only done a little bit of testing. 

The only small piece of feedback I have is that long feature names are cut off in the UI. Would it be possible to add a tooltip that displays the feature name when I hover over it? Not critical but it would be nice to have.

I also noticed what might be a bug, here are the repro steps:
1. Create a boolean flag with split targeting rules on a custom field. In my case the conditional is `EQUALS` and the type is `number`
2. Click into the custom split targeting rule and then click 'cancel'.

I saw some errors propagate in my dev console after performing those steps. Since the app is running in prod mode I didn't get a useful stacktrace but it looks like a null pointer: `Uncaught TypeError: this.a.c is null`. I don't think I would have noticed it if I didn't have my dev console open - it doesn't appear to impact functionality but like I said I haven't done extensive testing yet
